mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/MongoDBIndexOptionsConflictsTest.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.test.index;
+
+import static org.fest.assertions.Assertions.assertThat;
+import static org.hibernate.ogm.sessionfactory.SessionFactoryInlineTest.inTransaction;
+import static org.hibernate.ogm.sessionfactory.SessionFactoryInlineTest.insideCatchException;
+import static org.hibernate.ogm.sessionfactory.SessionFactoryInlineTest.withInSessionFactory;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+import org.hibernate.HibernateException;
+import org.hibernate.ogm.options.shared.IndexOption;
+import org.hibernate.ogm.options.shared.IndexOptions;
+import org.hibernate.ogm.sessionfactory.SessionFactoryConfig;
+import org.hibernate.ogm.utils.TestForIssue;
+
+import org.junit.Test;
+
+import com.mongodb.MongoCommandException;
+
+/**
+ * Tests what happens when an index and a constraint or different indexes are defined on the same column set.
+ *
+ * Some configurations are legal, because MongoDB could tolerate them, other not.
+ * It depends on the presence of index option conflicts.
+ * An index option conflict for MongoDB is when two indexes are defined on the same columns with different option configurations.
+ *
+ * Moreover a unique constraint is equivalent to a unique index. Then unique constraint can generate index option conflicts too.
+ * Even different types generate option conflicts.
```

Comment:
Is this sentence necessary considering the one above? If it is, it's unclear.

Reply:
The latter refers to conflicts generated using different types, whereas the former refers to conflicts generated using different unique attribute value.